src/lib/bootloader/proposal_client.rb

```diff
@@ -337,10 +343,28 @@ def handle_errors(ret)
 
     def single_click_action(option, value)
       stage1 = ::Bootloader::BootloaderFactory.current.stage1
-      devices = (option.to_sym == :mbr) ? stage1.boot_disk_names : stage1.boot_partition_names
+      case option
+      when "boot_mbr"
+        devices = stage1.boot_disk_names
+      when "boot_boot"
+        devices = stage1.boot_partition_names
+      when "trusted_boot"
+        ::Bootloader::BootloaderFactory.current.trusted_boot = value
+      when "secure_boot"
+        ::Bootloader::BootloaderFactory.current.secure_boot = value
+        if value && Yast::Arch.s390
```

Comment:
in validation it checks when there is a mismatch, but here it shows always when enabling. Why conditions are different?

Reply:
Because it is initially always off. Note also the different texts.